packages/holocron/src/holocronModule.jsx

```diff
@@ -108,36 +156,48 @@ export default function holocronModule({
 
     HolocronModuleWrapper.propTypes = {
       load: PropTypes.func,
+      dispatch: PropTypes.func.isRequired,
     };
 
     HolocronModuleWrapper.defaultProps = {
       load: undefined,
     };
 
-    HolocronModuleWrapper.displayName = WrappedComponent && `HolocronModule(${
-      WrappedComponent.displayName || WrappedComponent.name || name
-    })`;
+    HolocronModuleWrapper.displayName = getDisplayName(getName(WrappedComponent, name));
 
-    HolocronModuleWrapper[REDUCER_KEY] = reducer;
     if (load && (options.ssr || global.BROWSER)) {
       HolocronModuleWrapper[LOAD_KEY] = load;
     }
-    hoistStatics(HolocronModuleWrapper, WrappedComponent);
 
-    const getModuleState = createSelector(
-      (state) => state.getIn(
-        [MODULES_STORE_KEY, name],
-        reducer(undefined, { type: INIT_MODULE_STATE })
-      ),
-      (moduleState) => moduleState.toJS()
-    );
+    let mapModuleStateToProps;
+
+    if (reducer && name) {
+      HolocronModuleWrapper[REDUCER_KEY] = reducer;
+      const getModuleState = createSelector(
+        (state) => state.getIn(
+          [MODULES_STORE_KEY, name],
+          reducer(undefined, { type: INIT_MODULE_STATE })
+        ),
+        (moduleState) => moduleState.toJS()
+      );
+
+      mapModuleStateToProps = reducer && ((state) => {
+        const moduleState = getModuleState(state);
+        return { moduleState };
+      });
+    }
```

Comment:
Thinking about introducing a `console.warn` if reducer is provided but doesn't have a name. This may be better as a linting rule for Modules using `MyModule.holocron`.

Reply:
https://github.com/americanexpress/holocron/pull/21/files#r396855338